Framework/Kernel/test/VectorHelperTest.h

```diff
@@ -383,4 +385,107 @@ class VectorHelperTest : public CxxTest::TestSuite {
   std::vector<double> m_test_bins;
 };
 
+class VectorHelperTestPerformance : public CxxTest::TestSuite {
+public:
+  VectorHelperTestPerformance *createSuite() {
+    return new VectorHelperTestPerformance();
+  }
+  void destroySuite(VectorHelperTestPerformance *suite) { delete suite; }
+
+  VectorHelperTestPerformance() {
+    setupHistogram();
+    setupOutput();
+  }
+
+  void testRebinSmaller() {
+    auto size = smallerBinEdges.size() - 1;
+    for (size_t i = 0; i < nIters; i++) {
+      std::vector<double> yout(size);
+      std::vector<double> eout(size);
+      VectorHelper::rebin(binEdges, counts, errors, smallerBinEdges, yout, eout,
+                          false, false);
+    }
+  }
+
+  void testRebinSmallerFrequencies() {
+    auto size = smallerBinEdges.size() - 1;
+    for (size_t i = 0; i < nIters; i++) {
+      std::vector<double> yout(size);
+      std::vector<double> eout(size);
+      VectorHelper::rebin(binEdges, frequencies, frequencyErrors,
+                          smallerBinEdges, yout, eout, true, false);
+    }
+  }
+
+  void testRebinLarger() {
+    auto size = largerBinEdges.size() - 1;
+    for (size_t i = 0; i < nIters; i++) {
+      std::vector<double> yout(size);
+      std::vector<double> eout(size);
+      VectorHelper::rebin(binEdges, counts, errors, largerBinEdges, yout, eout,
+                          false, false);
+    }
+  }
+
+  void testRebinLargerFrequencies() {
+    auto size = largerBinEdges.size() - 1;
+    for (size_t i = 0; i < nIters; i++) {
+      std::vector<double> yout(size);
+      std::vector<double> eout(size);
+      VectorHelper::rebin(binEdges, frequencies, frequencyErrors,
+                          largerBinEdges, yout, eout, true, false);
+    }
+  }
+
+private:
+  const size_t binSize = 10000;
+  const size_t nIters = 10000;
+  std::vector<double> binEdges;
+  std::vector<double> counts;
+  std::vector<double> frequencies;
+  std::vector<double> errors;
+  std::vector<double> frequencyErrors;
+  std::vector<double> smallerBinEdges;
+  std::vector<double> largerBinEdges;
+
+  void setupHistogram() {
+    binEdges.resize(binSize);
+    frequencies.resize(binSize - 1);
+    frequencyErrors.resize(binSize - 1);
+    counts.resize(binSize - 1);
+    errors.resize(binSize - 1);
+
+    std::iota(binEdges.begin(), binEdges.end(), 0);
+    std::generate(counts.begin(), counts.end(),
+                  []() { return static_cast<double>(rand() % 1000); });
+
+    for (size_t i = 0; i < counts.size(); i++)
+      frequencies[i] = counts[i] / (binEdges[i + 1] - binEdges[i]);
+
+    std::transform(counts.cbegin(), counts.cend(), errors.begin(),
+                   [](const double count) { return sqrt(count); });
+
+    std::transform(frequencies.cbegin(), frequencies.cend(),
+                   frequencyErrors.begin(),
+                   [](const double freq) { return sqrt(freq); });
```

Comment:
Why all this init effort for performance tests? Shouldn't all zeros be just as good (apart from the bin edges)?